### DISCUSSION TITLE
Hide/show parts

### DISCUSSION THREAD

**Discussion by dboulanger363** [google/model-viewer#2860] [Q&A]
Is it possible to hide/show of parts of the model after loading using scene graph api for example?

**Comment by elalish** [comment]
Not yet, but that's a direction we would like to take it.

**Comment by loeffel-io** [comment]
+1

**Comment by onthez** [comment]
This is a feature I would love to see added as well

**Comment by milesgreen** [comment]
+1 👍

**Comment by sprengerst** [comment]
+1

**Comment by GuardianFree** [comment]
+1

**Comment by UnityEQ** [comment]
+1

**Comment by thang12l** [comment]
+1

**Comment by josephdave** [comment]
+1

**Comment by elalish** [comment]
So, this *is* actually possible, I just need to write an example for it. You need the part you want to hide to have its own glTF material, and set its `alphaMode` to `MASK` (`BLEND` would probably work too). Then in MV, once your object is loaded you can hide the object by setting it to transparent by setting the `baseColorFactor` to `[1, 1, 1, 0]`, and showing it again by resetting to `[1, 1, 1, 1]`. You can use [this example](https://modelviewer.dev/examples/scenegraph/#changeColor) as a guide. 

We should add a  helper function to select a material by name, but until then, you get find it by index or search the array for one with the right name.

**Comment by elalish** [comment]
We now have helper functions to select materials by name and by click: https://modelviewer.dev/docs/#entrydocs-scenegraph-methods-materialFromPoint

**Reply by NikLever** [reply]
I can only see materialFromPoint, no materialByName. I'm trying to hide an object by setting its material visibility to false but there doesn't appear to be ThreeJS type access to the object. You suggest setting baseColorFactor but that doesn't seem to be a material property.

**Reply by NikLever** [reply]
Sorted it
const material = modelViewer.model.getMaterialByName('CubeMat');
material.pbrMetallicRoughness.setBaseColorFactor([1,1,1,0]);

**Reply by slightHYP** [reply]
@NikLever I don't know if I understood well your issue, but have a look at this discussion [#2893](https://github.com/google/model-viewer/discussions/2893)

**Comment by sylvainlec** [comment]
+1
